src/shared/lib/GenericAssayUtils/MutationalSignaturesUtils.tsx

```diff
@@ -132,3 +133,23 @@ export function getSignificantMutationalSignatures(
             .value()
     );
 }
+
+export function validateMutationalSignatureRawData(
+    mutationalSignatureData: GenericAssayData[]
+): boolean {
+    const regex = new RegExp(
+        `(${MutationalSignatureStableIdKeyWord.MutationalSignatureContributionKeyWord}|${MutationalSignatureStableIdKeyWord.MutationalSignatureConfidenceKeyWord})`
+    );
+    const uniqueProfileIdGroupByVersion = _.chain(mutationalSignatureData)
```

Comment:
is the N here likely to be large?   if so, the following logic could certainly be optimized by using reduce and a dictionary to handle uniqueness.   right now you are looping through the whole thing 5 times when actually only once is necessary.

Reply:
updated with one round solution.